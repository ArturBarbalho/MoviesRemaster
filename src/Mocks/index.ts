import { BannerProps, CardProps, bigbannerProps } from "../typesProps";
export const BiggerBannerMock:bigbannerProps = {
  bigsrc: 'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
  smallsrc:'https://image.tmdb.org/t/p/original//ifRFLx83Xk1DcwAS3OScgI6HmWO.jpg',
  genres:['Action', 'Comedy'],
  id:2,
  media:'movie',
  overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
  title:'ONYANKOPON 3 O IN√çCIO'
}
export const CardMock:CardProps = {
  id:2,
  media:'movie',
  src:'http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg',
  title:'ONYANKOPON 2 O RETORNO'
}

export const BannerMock:BannerProps = {
    media: 'movie',
    relaseDate: '(1800)',
    id:1,
    title:'ONYANKOPON',
    src:'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
    overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
    genres: ['Action', 'Comedy' ]
  }