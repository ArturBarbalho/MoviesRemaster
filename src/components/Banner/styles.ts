import styled from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.main``

export const About = styled.div`
color:aliceblue;
position:absolute;
text-shadow:1px 1px black;
display:flex;
flex-direction:column;
align-items:center;
justify-content:center;
width: 100%;
height:100vh;
background-color:rgb(0,0,0,0.3);
${media.lessThan('small')`
    width:100%;
    overflow-x:hidden;
`}

`
export const Icon = styled.div`
display: flex;
justify-content:space-between;
width:350px;
${media.lessThan('small')`
width:100%;
   
`}
`