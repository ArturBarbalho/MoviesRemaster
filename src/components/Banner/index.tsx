import * as S from './styles'
import Image from '../Image/';
import Title from '../Title';
import Button from '../Button';
import { Functions } from '../../getdata';
import { BannerProps } from '../../typesProps/index';
import Overview from '../Overview';
type props = {
  bannerProps: BannerProps
}
const Banner = ({bannerProps}:props) => (
  <S.Wrapper>

    <S.About>
    
    <Title size='large' > {bannerProps.title}<span>{bannerProps.relaseDate}</span> </Title>
    <S.Icon>
          <img alt='icon1' src='icon1.png' />
          <img alt='icon2' src='icon2.png'/>
          <img alt='icon3' src='icon3.png'/>
          <img alt='icon4' src='icon4.png'/>
    </S.Icon>
    <Overview overview={bannerProps.overview} size='small' />
    <Button onClick={ ()=> Functions.Details(bannerProps.title,bannerProps.id,bannerProps.media)} color='black'  />
    </S.About>

    <Image src={bannerProps.src} size='large'/>
  </S.Wrapper>
)

export default Banner