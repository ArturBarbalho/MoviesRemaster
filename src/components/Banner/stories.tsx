import { Story, Meta } from '@storybook/react/types-6-0'
import Banner from '.'
import { BannerMock } from '../../Mocks'

export default {
  title: 'Banner',
  component: Banner,
  
} as Meta

export const Default: Story = () => <Banner bannerProps={BannerMock} />

