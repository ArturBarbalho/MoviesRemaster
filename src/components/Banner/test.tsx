import { render, screen } from '@testing-library/react'

import Banner from '.'
import { BannerMock } from '../../Mocks'

describe('<Banner />', () => {
  it('should render the image src correctly', () => {
    const { container } = render(<Banner bannerProps={BannerMock} />)

    expect(screen.getByAltText('icon1')).toHaveAttribute('src', 'icon1.png')
    expect(screen.getByAltText('icon2')).toHaveAttribute('src', 'icon2.png')
    expect(screen.getByAltText('icon3')).toHaveAttribute('src', 'icon3.png')
    expect(screen.getByAltText('icon4')).toHaveAttribute('src', 'icon4.png')

    expect(container.firstChild).toMatchSnapshot()
  })
})