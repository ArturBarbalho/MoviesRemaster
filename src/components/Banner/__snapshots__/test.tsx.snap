// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render the image src correctly 1`] = `
<main
  class="styles__Wrapper-sc-10z8pg5-0"
>
  <div
    class="styles__About-sc-10z8pg5-1 cdBpPv"
  >
    <h1
      class="styles__Wrapper-sc-1buurg8-0 kNhFRm"
    >
       
       
      ONYANKOPON
      <span>
        (1800)
      </span>
       
       
    </h1>
    <div
      class="styles__Icon-sc-10z8pg5-2 fXDdrQ"
    >
      <img
        alt="icon1"
        src="icon1.png"
      />
      <img
        alt="icon2"
        src="icon2.png"
      />
      <img
        alt="icon3"
        src="icon3.png"
      />
      <img
        alt="icon4"
        src="icon4.png"
      />
    </div>
    <p
      class="styles__Wrapper-sc-mhbed0-0 fSwUgG"
    >
      Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...
    </p>
    <button
      class="styles__Wrapper-sc-11qq8g6-0 hNDJSY"
      color="black"
    >
      See more
    </button>
  </div>
  <img
    class="styles__Wrapper-sc-y6e3xk-0 kFeMay"
    src="https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg"
  />
</main>
`;
